client/components/themenew/index.jsx

```diff
@@ -0,0 +1,12 @@
+import Theme from './theme';
+import { ThemeContextProvider } from './theme-context';
+
+import '../theme/style.scss';
+
+export default function ( { isPlaceholder = false, themeId = '', themePosition = 0 } ) {
+	return (
+		<ThemeContextProvider themeId={ themeId } themePosition={ themePosition }>
```

Comment:
Wouldn't it be more efficient to not create a new context provider for each theme and just have a single Context provider for the whole theme list instead?

Reply:
That's the Redux state? 😄 

The theme-specific context is a shortcut to propagate theme-specific info — both from Redux and internal state, like the selected variation.

I think it's an important distinction, also for readability purposes:
- Showcase context for list-wide details.
- Theme context for theme-specific ones.